### ISSUE TITLE
Best way to manage multiple transporters to multiple servers

### ISSUE THREAD

**Issue by ricardopolo** [nodemailer/nodemailer#639]
In our solution we send emails through multiple servers in parallel and in some cases using direct.

So we have a function where we get a new transporter and use it just when needed. In the function we just `return nodemailer.createTransport(options);` where options are the host and port of the desired server.

The issue that I am seeing is that sometimes we see emails to the wrong sever, that is, using another tranporter.  Also sometimes we receive `The maximum number of concurrent connections has exceeded a limit, closing transmission channel`. Not sure if that error is a related.

For me it looks like nodemailer is re using the transport or something, like in a pooled connection, But we havent configure it?

Any advice? Should we keep a tranporter per server in RAM? We may have lots of them, like 1000.
Or is it ok to generate it when needed? Why may be reusing them if `pool` options is not `true`?

Thanks in advance

**Comment by andris9** [comment]
I'd guess it is the nodemailer-direct module that tries to deliver deferred/multiple-recipient messages, not regular nodemailer transports re-using the same connection.

In general the described use case does not fit too well with the intended use case of Nodemailer. I would suggest you to skip Nodemailer and instead use [smtp-connection](https://github.com/nodemailer/smtp-connection) (+[mailcomposer](https://github.com/nodemailer/mailcomposer)) to have a more granular control of what is currently going on, at least that's what I do when I need to deliver a lot of messages to different hosts.

**Comment by ricardopolo** [comment]
@andris9 ok, i am planning to use mailcomposer with haraka to send messages. Using mailcomposer can I strill have plugins like odemailer-plugin-inline-base64 or nodemailer-html-to-text?

**Comment by ricardopolo** [comment]
Hi @andris9 sorry for asking this, because this is not related to Nodemailer. But knowing your deep knowledge about email and SMTP I think you can be the best person that can have a possible solution in this case.

I been trying with this issue in the last few months and could find a solution yet.
After your reply in this issue I changed the code to use Haraka for the sending part of our code and I am having exactly the same behavior. Mails from one server goes througth another one.

This is the issue in Haraka https://github.com/haraka/Haraka/issues/1556

In short this is what i found:

I send emails to multiple companies using multiple SMTP servers, one server from Company1 and one server from Company2, but sometimes emails from Company1 goes though Company2 servers

1. Company1 has this smtp server: Company1-protection.outlook.com 
2. Company2 has this smtp server: Company2-protection.outlook.com
3. Both DNS names resolves to the same IP.

These domains are the SMTP servers given for Microsoft for each Office 365 tentant.

When sending emails to Company 1 and then later sending emails to Company 2, emails from Company 2 goes through Company 1 servers. 
I don't know how Office 365 detects to which company is going an email if both names resolves to the same IP.

What could be causing that confusion in Office 365?
There is not pooling features configured in Haraka. And both servers don't require authentication.

Thank you so much in advance for any help!!  I will really appreciate that

**Comment by andris9** [comment]
What do you mean by "emails from Company2 goes through Company1 servers"? If it's the same IP then it's the same server (or load balancer) as well, no matter what the hostname is

**Comment by ricardopolo** [comment]
@andris9 Thanks for taking the time for answering this. I truly appreciate that! 👍 

I mean...
When the administrator  of Company 1 goes to its Office 365 Exchange console, you see emails from Company 2 and Company 3, etc in the logs. You see other companies emails going thought his infrastructure.

That is very bad for our product, you don't want info from one customer leaking into another customer logs 🔥 

I think, off course, Microsoft shares theirs servers across multiple customers,  but don't know what uses it to identify that a specific emails goes through a specific customer.
All MX servers that Microsoft give to each customer has that format (`customer-protection.outlook.com`) and resolve to a shared server. 

However you never see logs from a company into another one. However when sending to multiple customers in a few seconds using Nodemailer (or Haraka) logs are mixed!! I though it was something related to pooling but even disabling it continue having that behavior.

Regards!!
